packages/m2m-gateway/src/api/clientApiConverter.ts

```diff
@@ -14,17 +15,44 @@ export function toGetClientsApiQueryParams(
     offset: params.offset,
   };
 }
-export function toM2MGatewayApiConsumerClient(
-  client: authorizationApi.Client
-): m2mGatewayApi.Client {
+
+export function toM2MGatewayApiFullClient(
+  client: authorizationApi.FullClient
+): m2mGatewayApi.FullClient {
   assertClientKindIs(client, authorizationApi.ClientKind.Values.CONSUMER);
+
   return {
+    visibility: client.visibility,
     id: client.id,
-    consumerId: client.consumerId,
     name: client.name,
     createdAt: client.createdAt,
+    consumerId: client.consumerId,
     description: client.description,
-    purposes: client.purposes,
-    users: client.users,
   };
 }
+
+export function toM2MGatewayApiClient(
```

Comment:
minor point: since `API Client` exists, the name of this function may be a bit confusing

Reply:
Added a "Consumer" prefix